iroh/src/protocol.rs

```diff
@@ -109,10 +112,18 @@ pub struct RouterBuilder {
 /// The protocol handler must then be registered on the node for an ALPN protocol with
 /// [`crate::protocol::RouterBuilder::accept`].
 pub trait ProtocolHandler: Send + Sync + std::fmt::Debug + 'static {
+    /// Optional interception point to handle the `Connecting` state.
```

Comment:
I'm always a fan of being extremely verbose on the docs.  I would add a paragraph suggesting when you might to use this:
```
This enables accepting 0-RTT data from clients, among other things.
```